src/main/java/seedu/storemando/logic/parser/ReminderCommandParser.java

```diff
@@ -0,0 +1,39 @@
+package seedu.storemando.logic.parser;
+
+import static seedu.storemando.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import seedu.storemando.logic.commands.ReminderCommand;
+import seedu.storemando.logic.parser.exceptions.ParseException;
+import seedu.storemando.model.item.ItemExpiringPredicate;
+
+/**
+ * Parses input arguments and creates a new ReminderCommand object
+ */
+public class ReminderCommandParser implements Parser<ReminderCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the ReminderCommand
+     * and returns a ReminderCommand object for execution.
+     *
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    @Override
+    public ReminderCommand parse(String args) throws ParseException {
+        String trimmedArgs = args.trim();
+        if (trimmedArgs.isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, ReminderCommand.MESSAGE_USAGE));
+        }
+
+        String[] numOfArgs = trimmedArgs.split(" ");
+        if (numOfArgs.length > 1) {
```

Comment:
Perhaps, you can try checking both conditions for trimmedArgs and numOfArgs in one if block and throw a ParseException to avoid repetition?

Reply:
I feel that this shouldn't be an exception condition because we should instead just ignore whatever that's not the first arg. So reminder 6 haha should just take it as reminder 6. on the other hand reminder haha 6 shouldn't work but it has been considered in ur try catch method below.

Reply:
@Md-Fazil All right I will try to do so.
@JayChenYJ I personally feel it's not good to allow the user to input "reminder 6 hahahaha" as an acceptable input.